xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserSheet.xml

```diff
@@ -46,38 +46,23 @@
 ## Setting categories
 #########################
 #set($categories = [])
-#set($discard = $categories.add({'id':'profile', 'sheet':'XWiki.XWikiUserProfileSheet', 'glyphicon': 'user'}))
-#set($isMyProfile = ($services.model.resolveDocument($xcontext.user) == $doc.documentReference))
-#if($isMyProfile || $hasAdmin)
-  #set($discard = $categories.add({'id':'preferences', 'sheet':'XWiki.XWikiUserPreferencesSheet', 'glyphicon': 'wrench'}))
-#end
-#set($discard = $categories.add({'id':'groups', 'sheet':'XWiki.XWikiUserMembershipSheet', 'glyphicon': 'group'}))
-## TODO: add APIs to be able to display users watchlists to admins
-#if($isMyProfile &amp;&amp; $hasWatch)
-  #set($discard = $categories.add({'id':'watchlist', 'sheet':'XWiki.XWikiUserWatchListSheet', 'glyphicon': 'eye'}))
-#end
-#if($isMyProfile)
-  #if ("$!services.notification.watch" != '' || $hasWatch)
-    #set($discard = $categories.add({'id':'network', 'sheet':'XWiki.XWikiUserNetworkSheet', 'glyphicon': 'world'}))
-  #end
-  ## TODO: add an enhancement system instead
-  #set($notificationsSheet = $services.model.createDocumentReference('', ['XWiki', 'Notifications', 'Code'], 'XWikiUserNotificationsSheet'))
-  #if ($xwiki.exists($notificationsSheet))
-    #set($discard = $categories.add({'id':'notifications', 'sheet':'XWiki.Notifications.Code.XWikiUserNotificationsSheet', 'glyphicon': 'bell'}))
+
+## load the user menu from the 'org.xwiki.plaftorm.user.profile.menu.content' UIXP.
+#set($list_uix = ["org.xwiki.platform.user.profile.entry.userprofilesheet",
+  "org.xwiki.platform.user.profile.entry.userpreferencessheet",
+  "org.xwiki.platform.user.profile.entry.usermembershipsheet",
+  "org.xwiki.platform.user.profile.entry.userwatchlistsheet",
+  "org.xwiki.platform.user.profile.entry.usernetworksheet",
+  "org.xwiki.platform.user.profile.entry.usernotificationsheet",
+  "org.xwiki.platform.user.profile.entry.userwikisheet",
+  "org.xwiki.platform.user.profile.entry.userdashboardsheet"])
```

Comment:
I don't understand this static list. This defeats the concept of UIXP for me :)

Reply:
That not a closed list, just the list of UIX we know exist and want to keep sorted in the same order are they used to be before the refactoring.
See https://extensions.xwiki.org/xwiki/bin/view/Extension/UIExtension%20Module : 
> **sortByCustomOrder**: takes a list of UI Extension IDs as parameter. The extensions will be in the same order as they've been listed, additional extensions will be put at the end of the list.

An alternative would be to introduce a `priority` key in the UIX and to use it with the **sortByParameter** filter. WDYT?

Reply:
yes it's not good at all. We need the UIX to specify the order.

Reply:
The use case is that any UIX needs to be insertable at any position.

Reply:
The alternative is to provide a sort by some criteria but it's less easy since that could depend on the language. Needs to be decided. In any case it must not be hardcoded.

Reply:
I have introduced a `priority` property used in combination with the sortByCustomOrder parameter. The lower the priority is, the higher the UIX is in the menu.
Each existing UIX priority is a multiple of 10, that way external UIX can interleave between existing entries without having to modify "concurrent" UIX.

Reply:
Cool. Just a small detail: I would check how other UIXP have implemented priorities and check that the order is the same (lower is higher) so that we have some consistency ;)

Reply:
Seems to be the standard behaviour across the code base, and the way `sortByParameter` works.

For another example see https://github.com/xwiki/xwiki-platform/blob/07c0927c9bf1a6a48db4ef8deb286f16d6e7c1b7/xwiki-platform-core/xwiki-platform-panels/xwiki-platform-panels-ui/src/main/resources/PanelsCode/ApplicationsPanelConfigurationSheet.xml#L71